### PR TITLE
[G7T5-45] Job Skill API and Unit Tests

### PR DESCRIPTION
**Pull request by deadline2359**
# Description

Created Job Skill Table and Unit Tests

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by quinncheong — APPROVED** [review]

File: backend/app/api/api_v1/endpoints/job_skill.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        raise HTTPException(
+            status_code=404,
+            detail="Jobs and skills not found",
+        )
+    return job_skill
+
+
+@router.get("/jobs/{job_id}", response_model=List[schemas.JobSkill])
+def get_job_skill_by_job_id(
+    job_id: int, db: Session = Depends(deps.get_db), skip: int = 0
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job_skill.py:32]: /nit should this be in different lines?

File: backend/app/api/api_v1/endpoints/job_skill.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        raise HTTPException(
+            status_code=404,
+            detail="Skills under this job not found",
+        )
+    return job_skill
+
+
+@router.get("/skills/{skill_id}", response_model=List[schemas.JobSkill])
+def get_job_skill_by_skill_id(
+    skill_id: int, db: Session = Depends(deps.get_db), skip: int = 0
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job_skill.py:48]: /nit should this be in different lines?

File: backend/app/api/api_v1/endpoints/job_skill.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+#         raise HTTPException(
+#             status_code=404,
+#             detail="Combination of Job and Skill not found",
+#         )
+#     job_skill_in = schemas.JobSkillCreate(
+#         skill_id=job_skill.skill_id,
+#         job_id=job_id or job_skill.job_id,
+#     )
+#     job_skill = crud.job_skill.update(db, db_obj=job_skill, obj_in=job_skill_in)
+#     return job_skill
```
Comment (jonathantan1425) [inline backend/app/api/api_v1/endpoints/job_skill.py:106]: Are we planning to keep this for future reference? Else we can just delete

File: backend/app/crud/crud_job_skill.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    #     db: Session,
+    #     *,
+    #     db_obj: Job_Skill,
+    #     obj_in: Union[JobSkillUpdate, Dict[str, Any]],
+    # ) -> Job_Skill:
+    #     if isinstance(obj_in, dict):
+    #         update_data = obj_in
+    #     else:
+    #         update_data = obj_in.dict(exclude_unset=True)
+    #     return super().update(db, db_obj=db_obj, obj_in=update_data)
```
Comment (jonathantan1425) [inline backend/app/crud/crud_job_skill.py:79]: Do we wanna keep this for future reference? Else I think we can just delete it

File: backend/app/tests/api/api_v1/test_job_skill.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+# def test_update_job_skill_by_id_non_existent(client: TestClient, db: Session) -> None:
+#     data = {"job_id": 123, "skill_id": 456}
+#     response = client.put(
+#         f"{settings.API_V1_STR}/job_skill/999999",
+#         json=data,
+#     )
+#     assert response.status_code == 404
+#     content = response.json()
+#     assert content["detail"] == "Job and its skills not found"
```
Comment (jonathantan1425) [inline backend/app/tests/api/api_v1/test_job_skill.py:150]: Do we want to keep this for future reference? Else we can just delete it

**Review by jonathantan1425 — APPROVED** [review]
LGTM other than a couple of nits

**Pull request merged**